File: DS/排序算法/bubble_sort.cpp
/*
 * @Author: ykdsir
 * @Date: 2020-06-20 08:32:18
 * @LastEditTime: 2020-06-21 17:55:33
 * @LastEditors: ykdsir
 * @Description: 
 * @FilePath: /algorithms/DS/sort/bubble_sort.cpp
 * @
 */ 

#include <iostream>
#include <vector>
using namespace std; 

void BubbleSort(vector<int>& arr, int n) {
    for(int i = 0; i < n; ++i){
        bool flag = true;
        for(int j = 0; j < n-i-1; ++j){
            if(arr[j] > arr[j+1]){
                swap(arr[j], arr[j+1]);
                flag = false;
            }
        }
        if(flag){
            break;
        }
    }   
}

int main(){
    vector<int> arr = {5,5,6,5,4,6,1,4,2,3,7};
    BubbleSort(arr, arr.size());
    for(int i = 0; i < arr.size(); ++i){
        cout << arr[i] << " ";
    }
    cout << endl;
}

File: DS/排序算法/bubble_sort.py
'''
@Author: ykdsir
@Date: 2020-06-20 08:39:42
@LastEditTime: 2020-06-20 08:43:51
@LastEditors: ykdsir
@Description: 
@FilePath: /algorithms/DS/sort/bubble_sort.py
@
'''

def bubble_sort(arr, n):
    for i in range(n):
        for j in range(n-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]


if __name__ == "__main__":
    arr = [5,5,5,5,4,4,4,4,6,1,2,3,7,1]
    res = sorted(arr)
    bubble_sort(arr, len(arr))
    print(arr)
    assert res == arr